tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -210,17 +213,21 @@ public:
    void Read(NTupleSize_t globalIndex, RFieldValue *value) {
       if (!fIsSimple) {
          ReadGlobalImpl(globalIndex, value);
-         return;
+      } else {
+         fPrincipalColumn->Read(globalIndex, &value->fMappedElement);
       }
-      fPrincipalColumn->Read(globalIndex, &value->fMappedElement);
+      if (fReadCallback)
```

Comment:
I think a key implementation point of this feature is to avoid an additional `if` on every read to check for the read callback.  The fast path should only need to check `fIsSimple`. So, if we have a read callback, we wouldn't consider the field "simple" anymore.

Looking at #11628, we could have a `kTraitMappable` trait that tells us if the *type* has the requirements to qualify for being simple. Additional conditions must then be true to mark the field as simple (namely: not having a read callback).